RA-222: Login button should be enabled at all times

ivange94 opened the pull request:
https://issues.openmrs.org/browse/RA-222

ivange94 commented:
i think this is good to go.

dkayiwa commented on omod/src/main/webapp/pages/login.gsp:
```diff
@@ -41,6 +37,15 @@ ${ ui.includeFragment("referenceapplication", "infoAndErrorMessages") }
+        jQuery('#login-form').submit(function(e) {
+        	var sessionLocationVal = jQuery('#sessionLocationInput').val();
+        	
+        	if (!sessionLocationVal) {
```
@ivange94 did you intentionally leave out some of the tests that the line 27 above does?

ivange94 replied:
@dkayiwa i don't understand the question. Line 29 should always be executed when updateSelectedOption() is called, the condition makes sure that a location was actually chosen. Is that what you were asking?

dkayiwa replied:
Line 28 has three checks to confirm that location is not selected. Your change has one check. Did you just ignore the other two checks?

ivange94 replied:
@dkayiwa Oh i see. I didn't think it was necessary to do all the 3 checks. Just the one is enough. Because in javascript, null, empty string and 0 are falsy which means when you use them in place of a boolean they always return false. So there was no need doing all the checks separately. Does that make sense?

dkayiwa replied:
@ivange94 if sessionLocationVal == "", will your test catch it?

ivange94 replied:
Yes. "" returns false.  hence !sessionLocationVal, where sessionLocationVal == "" will return true.

dkayiwa commented on omod/src/main/compass/sass/login.scss:
```diff
@@ -8,4 +8,9 @@
+}
+
+.location-error {
+	color: red;
```
Wouldn't the existing class "field-error" work?

ivange94 replied:
i didn't know there was an existing error class. I have looked at it, it would probably work fine. I am testing it now. I have not seen the css file where the styles for field-error are declared though. I have seen just the referenceapplication.css, but this is generated at runtime and git is set to ignore these. Can you please point me to the .scss file where field error is declared, in case i need to add some more styles? :)

dkayiwa replied:
https://github.com/openmrs/openmrs-module-uicommons/blob/b0873ce7c2231a9cfd51e31eb52fd1b305edb578/scss/src/main/resources/sass/reference/fragments/_form-navigator-ui.scss#L58

ivange94 replied:
Thanks. Am building and launching the ref app now to test. will push changes in a few minutes. :)

ivange94 replied:
The style was not applied. Am guessing its because i have not actually included the css file inside the logn.gsp.
To include that css file, am suppose to use something like ui.includeCss("uicommon", "fileName.css"), correct? If yes which file should i replace with fileName.css. If no, how to include the generated css inside login.gsp.

I thought about this because login.css is included using ui.includeCss("referenceapplication", "login.css") so am guessing am suppose to do something similar but replace "referenceapplication" with "uicommons" but i don't know which generated css file should i replace "login.css" with. I don't have enough internet data now so i can't clone uicommons and build and look for the generated file that will hold the .field-error. I will probably do that tomorrow when i have bought some internet data. But if you know which file it is am suppose to include, you can just save me the trouble and lets finish with this ticket today. :)

ivange94 commented on omod/src/main/webapp/pages/login.gsp:
```diff
@@ -1,6 +1,7 @@
 <%
     ui.includeFragment("appui", "standardEmrIncludes")
     ui.includeCss("referenceapplication", "login.css")
+    ui.includeCss("uicommons", "form-navigator-ui.css");
```
@dkayiwa i tried to follow the pattern i saw in other files and tried including form-navigator-ui.css even thought there is only a form-navigator-ui.scss, but it still did not work, as a result, the style specified for .field-error is not applied. :(

ivange94 commented:
@dkayiwa i finally figured out why the style specified for .field-error was not been applied. Looking at [here](https://github.com/openmrs/openmrs-module-uicommons/blob/master/scss/src/main/resources/sass/reference/fragments/_form-navigator-ui.scss#L58) i did not notice that .field-error was actually nested inside simple-form-ui. field-error class could only work if its put inside a parent .simple-form-ui parent, so i added .simple-form-ui to the form element on the page. Also the display type for .field-error was specified as none hence making the error message to be displayed under the location notification like [here](http://picpaste.com/pics/block_error-81nhxa1z.1460941938.png) so i changed it to inline so that it appears on the same line like [here](http://picpaste.com/pics/with_inline_display-z0v53YxQ.1460942709.png). Considering that .field-error is supposedly used elsewhere, i don't know how that change is going to affect the other use cases. Please look at the photos and tell me if should keep the block display or use the inline display.

dkayiwa commented:
@ivange94 in that case it is safer to just ignore field-error and use your own.

ivange94 commented:
@dkayiwa done. Final [screenshot](http://picpaste.com/pics/using_location_error-WZmVo2lq.1460972955.png) (its exactly the same as that on jira)
